wit/deps/buffer-pool/buffer-pool.wit

```diff
@@ -89,8 +90,13 @@ interface buffer-pool {
         /// name: the name of the pool. you can use this for device.start().
         create: static func(mode:buffering-mode, size:u32, buffer-num:u32, name:string) ->result<pool, buffer-error>;
 
-        block-read-frame: func()-> result<frame-info, buffer-error>;
-        poll-read-frame: func()-> result<frame-info, buffer-error>;
+        /// try to read frames.
+        /// this function returns 0 frames when
+        ///  - max-results = 0
+        ///  - or, no frames are immediately available
+        read-frames: func(max-results: u32)-> result<list<frame-info>, buffer-error>;
```

Comment:
i don't understand what you mean.
read-frames here is non-blocking.

blocking read can be easily implemented with subscribe and read-frames. see the guest code.

Reply:
Could you also write the sample guest code to show how to create a callback function using read-frames ?

Reply:
> Could you also write the sample guest code to show how to create a callback function using read-frames ?

what's the point?

Reply:
I should have asked you to write a test guest code instead of a sample code.
Currently, it seems there is only test code for block read.

Reply:
> I should have asked you to write a test guest code instead of a sample code. Currently, it seems there is only test code for block read.

what exactly do you meat by "block read"?

Reply:
https://github.com/midokura/wasi-sensor-discussion/blob/62befe18738df02e51d783842bfdbb46d4aa61ca/guest_c/main.c#L167-L174

Reply:
wasi_io_0_2_0_rc_2023_11_10_poll_method_pollable_block blocks. (just like poll system call on unix)
wasi_buffer_pool_buffer_pool_method_pool_read_frames doesn't.

Reply:
>  Well, you can say that you are testing it.

???